tests/event_loop_test.c

```diff
@@ -1070,7 +1071,7 @@ AWS_TEST_CASE(event_loop_group_setup_and_shutdown, test_event_loop_group_setup_a
 /* mark the thread complete when the async shutdown thread is done */
 static void s_async_shutdown_thread_exit(void *user_data) {
     struct task_args *args = user_data;
-    args->thread_complete = true;
+    aws_atomic_store_int(&args->thread_complete, true);
```

Comment:
the lock is always acquired in the predicate, and unlocked during wait. why does this need to be atomic? You can just lock in the callback.

Reply:
It's an ordering violation: we need to ensure the cleanup thread exit happens-before the test finishes. A lock by itself won't do. Do you mean use a condition variable? That would work, but this use of an atomic is simpler, I think.